File: proofOfAccount/generator.py
import secrets
import string

def generate_unique_random_string(length=32):
    characters = string.ascii_letters + string.digits
    while True:
        random_string = ''.join(secrets.choice(characters) for _ in range(length))
        if random_string not in generated_strings:
            generated_strings.add(random_string)
            return random_string

generated_strings = set()
print(generate_unique_random_string())

File: proofOfAccount/proof-of-account/src/Generator.js
const generatedStrings = new Set();

function generateUniqueRandomString(length = 32) {
  const characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
  while (true) {
    let randomString = '';
    for (let i = 0; i < length; i++) {
      randomString += characters.charAt(Math.floor(Math.random() * characters.length));
    }
    if (!generatedStrings.has(randomString)) {
      generatedStrings.add(randomString);
      return randomString;
    }
  }
}

console.log(generateUniqueRandomString());
